### PR TITLE
Decorate class

### PR DESCRIPTION
**Pull request by Arturgouveia1970**
In this PR I added decorator classes: 
- decorator
- capitalize_decorator
-  trimmer_decorator
-  nameable

### CODE REVIEW

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative './nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
```
Comment (Deepakdanger) [inline decorator.rb]: - [ ] Kindly include `attr_accessor :nameable ` after class declaration.

File: trimmer_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require './decorator'
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name[0...10]
```
Comment (Deepakdanger) [inline trimmer_decorator.rb]: - [ ] Here we have to trim only when the length is greater than 10.Kindly use 
`@nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name`

File: capitalize_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require './decorator'
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.Capitalize
```
Comment (Deepakdanger) [inline capitalize_decorator.rb]: - [ ] I think this will give you an error. Kindly use `capitalize` instead of `Capitalize`.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @Arturgouveia1970, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 🍷 

## Hi @Arturgouveia1970 👋, 

Well done implementing all the required changes 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights 🎯 
✔️ Descriptive PR
✔️ No linter errors

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**